### ISSUE TITLE
Packager crashing on publish when uploading assets

### ISSUE THREAD

**Issue by hashim-sohail** [expo/expo-cli#1091]
## Description

Getting `Maximum call stack size exceeded` on assets when running `expo publish`

## Expected Behavior
To publish app on expo

## Observed Behavior
It throws error on assets when bundling up for IOS and Android

```
[10:25:34] Starting Metro Bundler on port 19001.
[10:25:38] Tunnel ready.
[10:25:38] Publishing to channel 'feature-github-workflow'...
[10:25:39] Building iOS bundle
[10:29:04] assets/common/logo.png: Maximum call stack size exceeded
[10:29:04] › Closing Expo server
[10:29:04] assets/common/logo.png: Maximum call stack size exceeded
[10:29:04] Failed building JavaScript bundle.
[10:29:04] › Stopping Metro bundler
[10:29:04] Packager URL http://127.0.0.1:19001/build/expo/AppEntry.bundle?dev=false&minify=true&hot=false&platform=ios returned unexpected code 500. Please open your project in the Expo app and see if there are any errors. Also scroll up and make sure there were no errors or warnings when opening your project.
```

## Environment
Running in github actions, getting same error on Ubuntu 18.04

## Reproducible Demo
It is a private project

## Logo File
![logo](https://user-images.githubusercontent.com/22448789/66487020-14cdcf80-eac5-11e9-809f-058140773954.png)

**Comment by lrholmes** [comment]
Getting this same error, but on `node_modules/react-navigation-stack/lib/module/views/assets/back-icon-mask.png`

Did you make any progress on this @hashim-sohail?

**Comment by hashim-sohail** [comment]
@lrholmes Nope. Stuck on this issue, getting the same errors in multiple applications.

Were you able to find something?

**Comment by lrholmes** [comment]
If you make a fresh Expo install and change only the icon from default to yours, does the issue persist? I wonder if that's the real root cause or not.

I wasn't able to solve the issue directly, but I made a fresh Expo app, and almost file-by-file introduced elements of my project (starting with app.json, package.json, babel config etc, then moving onto screens and components), and built it to see if anything in particular broke it. I was able to add everything without it breaking. It's bizarre.

My native app shares code with a web app, which itself is part of a larger repo, so I now have to maintain two projects at once. It's frustrating but at the moment is working for me, as I only need to make small adjustments to the native app.

**Comment by brentvatne** [comment]
if someone can share a minimal reproducible example for this i would love to investigate. if you're unable to remake a minimal example please add me to a repo, private or otherwise, and i can try to look at that

**Comment by lrholmes** [comment]
Hey @brentvatne, I actually solved the issue for myself in the end and it's very strange. I am not able to create a repro atm but maybe describing the issue will help. Unfortunately I think it's probably unrelated to @hashim-sohail's one.

When starting my expo app, I was adding it to an existing project, monorepo consisting of a nextjs app, a server, and a couple other directories. The structure looked something like this after I added the expo project:

```
server/
app/
native/ # the expo dir
```

I quickly realised that I wanted to start sharing code from the existing next.js app, so moved the expo project so that it was nested inside `app/`. I think I did this by dragging and dropping the folder inside Finder.

```
 server/
 app/
  native/
```

When it came to finally start building the app, I started running into the above issue, and as stated I could only fix it by copying everything needed across to a new project. Keeping them in sync was a pain, but it was a valid solution for me to unblock the build failures.

Fast forward a little bit, and when looking at the root of my github repo, I noticed something strange. There was still a `native` folder listed. I clicked into it and there remained an `.expo` folder, which in turn had a `settings.json` and `package-info.json` file inside it. This was strange because the `native` folder at the root didn't exist as far as I could see, on my machine (maybe I overlooked that, not sure anymore). 
Anyway, I deleted the files via the Github interface, then pulled the changes down, and then was able to build again.

So the issue seemed to be caused by a stray `.expo` folder in the codebase somewhere. Would that make any sense?

**Comment by brentvatne** [comment]
oh that is interesting.. i'm just taking over this repo while @fson is on paternity leave so i don't have a good intuition on whether that would be the source of the problem. hopefully @hashim-sohail can provide some additional context to help us get to the bottom of this

**Comment by fson** [comment]
Closing this issue because it was opened a long time ago and it's going to be difficult to reproduce the bug.

If you still encounter this problem, please open a new issue and include a link to a GitHub repo (or similar) that reproduces the issue.
